shared/datastore_ae.go

```diff
@@ -92,6 +96,20 @@ func (d aeDatastore) Put(key Key, src interface{}) (Key, error) {
 	return datastore.Put(d.ctx, key.(*datastore.Key), src)
 }
 
+func (d aeDatastore) PutMulti(keys []Key, src interface{}) ([]Key, error) {
+	cast := make([]*datastore.Key, len(keys))
+	for i := range keys {
+		cast[i] = keys[i].(*datastore.Key)
+	}
+
+	cast, err := datastore.PutMulti(d.ctx, cast, src)
+	newKeys := make([]Key, len(cast))
+	for i := range cast {
+		newKeys[i] = cast[i]
```

Comment:
Nit: let's not reuse the variable to avoid confusion.

```suggestion
	srcKeys, err := datastore.PutMulti(d.ctx, cast, src)
	newKeys := make([]Key, len(srcKeys))
	for i := range srcKeys {
		newKeys[i] = srcKeys[i]
```